gateway/session_manager.go

```diff
@@ -133,21 +163,19 @@ func (l *SessionLimiter) limitRedis(currentSession *user.SessionState, key strin
 func (l *SessionLimiter) limitDRL(currentSession *user.SessionState, key string, rateScope string,
 	apiLimit *user.APILimit, dryRun bool) bool {
 
-	// In-memory limiter
-	if l.bucketStore == nil {
-		l.bucketStore = memorycache.New()
-	}
-
 	bucketKey := key + ":" + rateScope + currentSession.LastUpdated
 	currRate := apiLimit.Rate
 	per := apiLimit.Per
 
+	tokenValue := uint(l.drlManager.CurrentTokenValue())
+
 	// DRL will always overflow with more servers on low rates
-	rate := uint(currRate * float64(l.Gw.DRLManager.RequestTokenValue))
-	if rate < uint(l.Gw.DRLManager.CurrentTokenValue()) {
-		rate = uint(l.Gw.DRLManager.CurrentTokenValue())
+	cost := uint(currRate * float64(l.drlManager.RequestTokenValue))
```

Comment:
```suggestion
	cost := uint(currRate * float64(tokenValue))
```

Reply:
Current/Request token values are different values. It's correct and non-repeating.